packages/react-core/src/components/HelperText/HelperTextItem.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/HelperText/helper-text';
+import { css } from '@patternfly/react-styles';
+import MinusIcon from '@patternfly/react-icons/dist/js/icons/minus-icon';
+import ExclamationTriangleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-triangle-icon';
+import CheckIcon from '@patternfly/react-icons/dist/js/icons/check-icon';
+import TimesIcon from '@patternfly/react-icons/dist/js/icons/times-icon';
+
+export interface HelperTextItemProps extends React.HTMLProps<HTMLDivElement | HTMLLIElement> {
+  /** Content rendered inside the helper text item. */
+  children?: React.ReactNode;
+  /** Additional classes applied to the helper text item. */
+  className?: string;
+  /** Sets the component type of the helper text item. */
+  component?: 'div' | 'li';
+  /** Variant styling of the helper text item. */
+  variant?: 'default' | 'indeterminate' | 'warning' | 'success' | 'error';
+  /** Icon prefixing the helper text. This property will override the default icon paired with a dynamic helper text. */
+  icon?: React.ReactNode;
+  /** Flag indicating the helper text item is dynamic. */
+  isDynamic?: boolean;
+  /** Flag indicating the dynamic helper text should have an icon */
+  hasIcon?: boolean;
+}
+
+const variantStyle = {
+  default: '',
+  indeterminate: styles.modifiers.indeterminate,
+  warning: styles.modifiers.warning,
+  success: styles.modifiers.success,
+  error: styles.modifiers.error
+};
+
+export const HelperTextItem: React.FunctionComponent<HelperTextItemProps> = ({
+  children,
+  className,
+  component = 'div',
+  variant = 'default',
+  icon,
+  isDynamic,
+  hasIcon = true,
+  ...props
+}: HelperTextItemProps) => {
+  const Component = component as any;
+  return (
+    <Component
+      className={css(styles.helperTextItem, variantStyle[variant], isDynamic && styles.modifiers.dynamic, className)}
+      {...props}
+    >
+      {icon && (
```

Comment:
So the `icon` vs `hasIcon` interface still does not make sense to me.  I don't understand why the static helper text can have an icon if it is custom but not dynamic?  It is confusing that the consumer needs to provide an icon when the helper text is static.

@mcoker my understanding is that dynamic requires an icon but for that static it is optional.  is t hat correct?

Reply:
`icon` is an optional property that inserts an icon prefixing a helper text (affects both static and dynamic helper text, its just dynamic has default icons that this property would override). 
`hasIcon` is an optional property that controls whether the dynamic default icon is present (in cases where the included dynamic icon isn't desired).

Would `hasDynamicIcon` make more sense instead, or should static helper text also include the default icons?

Reply:
> my understanding is that dynamic requires an icon but for that static it is optional. is t hat correct?

@tlabaj correct, since the text color is the same between multiple states in the dynamic variation, the icon is used to convey the state (success, warning, etc).

Reply:
> Would `hasDynamicIcon` make more sense instead

I think so, assuming we want to keep the icon defaults as they are - regular has an optional icon that the user specifies, and dynamic comes with a specific icon you can either change or disable.

> should static helper text also include the default icons?

@mcarrano wdyt about this? I think this could be good in enforcing some consistency with icons used for the various states of helper text, though I don't know if the default use case for helper text should include an icon or not. I do know regular helper text (black text) wouldn't have an icon.

Another option could be to have `hasIcon` as a boolean and it's false for regular helper text and true for dynamic. Setting it to true with regular helper text would enable the default icons, and the defaults could be overridden by an `icon` prop (passing an `icon` prop would also just add an icon without having to set `hasIcon`). There would still be the use case of regular help text - what icon would `hasIcon` use? As far as I've seen, we haven't added an icon to regular helper text.

Reply:
The only thing I dislike about `hasDynamicIcon` to me, is that it implies the icon changes in some way. There might be a better name for the variable. Though if we end up going the route of including default icons on the static helper text we can keep `hasIcon`.

Another alternative is to get rid of `hasIcon` altogether, and have the component remove dynamic helper text icons if it is explicitly passed `icon={null}`?

Reply:
>  should static helper text also include the default icons?

I'm looking at the current examples we have for Forms and the default help text does not have an icon (https://www.patternfly.org/v4/components/form/html#help-text). I think we should keep it that way and let the user add an icon only when needed. Is that you question, @mcoker ? Or are you just asking whether the default status incons should be available in both versions? If the later, I agree that might lend to consistency. But the default for static helper text should still be no icon.

Does that make sense or did I confuse things more? :)

Reply:
> I think we should keep it that way and let the user add an icon only when needed. Is that you question, @mcoker

@mcarrano yep!

Reply:
Should I update `hasIcon` to work for static helper text then, leaving its default to `false`? And would the default icons be the same for each variant as the dynamic helper text?

Reply:
@kmcfaul Yes. Please make `hasIcon` valid for both versions but defaulting to false for the static variant. Regarding what default set of icons should be, I agree that they should be consistent. @mcoker now it looks like there are different icons for static and dynamic variants in the examples. Am I seeing that right? Do you know why these are different? Is it just that the designers gave you different icons?

Reply:
Followed up in slack: icons across both static/dynamic will match dynamic default icons for the moment. Might be updated later.